### PR TITLE
Fix broken '--node-selector' for osc new-project

### PR DESCRIPTION
**Pull request by pravisankar**
Handle osc new-project <name> --node-selector=""(empty selector)

### CODE REVIEW

**Comment by pravisankar** [comment]
Yesterday night I was preparing setup for demo and I ran into these 2 issues:
(a) --node-selector flag was ignored by project request resource.
(b) osc new-project <name> --node-selector="" didn't work as expected
This PR fixes (a) and (b)
@derekwaynecarr @deads2k please review/merge

**Comment by pravisankar** [comment]
[test]

File: pkg/cmd/admin/project/new_project.go [hunk 1]
```diff
@@ -38,6 +38,8 @@ an admin user (and role, if you want to use a non-default admin role), and a nod
 to restrict which nodes pods in this project can be scheduled to.
 `
 
+var CheckNodeSelector bool
```
Comment (smarterclayton) [inline pkg/cmd/admin/project/new_project.go]: No global variables

File: pkg/cmd/cli/cmd/request_project.go [hunk 2]
```diff
@@ -45,6 +45,8 @@ After your project is created you can switch to it using %[2]s <project name>.`
   $ %[1]s web-team-dev --display-name="Web Team Development" --description="Development project for the web team." --node-selector="env=dev"`
 )
 
+var CheckNodeSelector bool
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/request_project.go]: No global variables

File: pkg/cmd/cli/cmd/request_project.go [hunk 3]
```diff
@@ -107,7 +114,9 @@ func (o *NewProjectOptions) Run() error {
 	projectRequest.DisplayName = o.DisplayName
 	projectRequest.Annotations = make(map[string]string)
 	projectRequest.Annotations["description"] = o.Description
-	projectRequest.Annotations["openshift.io/node-selector"] = o.NodeSelector
+	if CheckNodeSelector {
+		projectRequest.Annotations["openshift.io/node-selector"] = o.NodeSelector
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/request_project.go]: This should be a constant in the node selector code and referenced here.

**Comment by smarterclayton** [comment]
End users can't request a node selector.  That means a non-admin can completely break a project.  Node selector has to be set by admin during project template creation.  Since that's effectively the default, there's no reason to set that.

**Pull request closed without merging**